### ISSUE TITLE
Experimental design not always showup in DEA heatmaps

### ISSUE THREAD

**Issue by ppavlidis** [PavlidisLab/Gemma#312]
Example https://gemma.msl.ubc.ca/expressionExperiment/showExpressionExperiment.html?id=23429

Clicking on one of the DEA icons yields a heatmap that has at least two problems: first, only one gene is shown even when there are many genes DE. Second, the experimental design isn't shown. 

I don't know if this is happening everywhere.

<img width="565" alt="image" src="https://user-images.githubusercontent.com/206415/157947917-02cc4238-dc01-4fbc-8338-ffc97e8d2ad7.png">

**Comment by ppavlidis** [comment]
On production, I still am seeing this happening: the design doesn't always show up, and only one gene is shown.

https://gemma.msl.ubc.ca/expressionExperiment/showExpressionExperiment.html?shortName=GSE215099 for example.

I also know there occasionally is an issue where the samples are not shown in the ordering they are supposed to (relative to the experimental design), but it will sometimes resolve itself. I thought this had been noted in an issue but I can't find it.

I suspect there is some sneaky bug, and that code it a bit of a hornet's nest.

**Comment by ppavlidis** [comment]
I have fixed the one-gene problem: it was caused by datavector valueobjects not having unique IDs.

I also may have fixed the disappearing experimental design.

I may have also fixed the problem with the scrambled vectors. I was able to reproduce this routinely and it seems to be a problem with cached vectors (value objects). I reworked some of ProcessedExpressionDataVectorDao.getProcessedDataArraysByProbeIds which was actually not using the cache correctly. At this point I can't reproduce it though I don't fully understand what was going on.
